framework/src/multiapps/TransientMultiApp.C

```diff
@@ -395,8 +445,61 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
             }
           }
         }
-        else if (!ex->lastSolveConverged())
-          throw MultiAppSolveFailure(name() + " failed to converge");
+        else
+        {
+          if (!ex->lastSolveConverged())
+          {
+            // Even if we don't allow auto_advance - we can still catch up to the current time if
+            // possible
+            if (_catch_up)
+            {
+              _console << "Starting Catch Up!" << std::endl;
+
+              bool caught_up = false;
+
+              unsigned int catch_up_step = 0;
+
+              Real catch_up_dt = dt / 2;
+
+              while (!caught_up && catch_up_step < _max_catch_up_steps)
+              {
+                Moose::err << "Solving " << name() << " catch up step " << catch_up_step
+                           << std::endl;
+                ex->incrementStepOrReject();
+
+                ex->computeDT();
+                ex->takeStep(catch_up_dt); // Cut the timestep in half to try two half-step solves
+
+                // This is required because we can't call endStep() yet
+                // (which normally increments time)
+                Real current_time = ex->getTime() + ex->getDT();
+
+                if (ex->lastSolveConverged())
+                {
+                  if (current_time + app_time_offset + ex->timestepTol() * std::abs(current_time) >=
```

Comment:
Is the order of operations correct here? (e.g. current_time + app_time_offset + (tol * current_time))? It just looks a little funny.

Reply:
It is right.

`current_time + app_time_offset` is the current time... then we're adding a (relative) little bit to it as a fudge and seeing if that's `>=` the `target_time`.

I added parens just to make it clearer